packages/myst-cli/src/transforms/code.ts

```diff
@@ -194,3 +194,45 @@ export function propagateBlockDataToCode(session: ISession, vfile: VFile, mdast:
     if (outputNode && !outputNode.visibility) outputNode.visibility = 'show';
   });
 }
+
+/**
+ * Because jupytext style notebooks can introduce code-cells in blocks
+ * after explicit thematic breaks, we need to lift them up to the parent
+ * to ensure that all executable code blocks are at the top level of
+ * the document.
+ *
+ * @param mdast
+ */
+export function transformLiftCodeBlocksInJupytext(mdast: GenericParent) {
+  const flattened = (mdast.children as GenericParent[]).reduce((acc, node) => {
+    if (node.type !== 'block' || !node.children) return [...acc, node];
+
+    const buriedCodeCells = selectAll('block:has(block)', node);
+    if (buriedCodeCells.length === 0) return [...acc, node];
+
+    const newBlocks: GenericParent[] = [{ type: 'block', children: [] }];
```

Comment:
Maybe this should be, like
```suggestion
    const newBlocks: GenericParent[] = [{ ...node, children: [] }];
```

to maintain metadata. (it would then be lost on the subsequent broken up blocks - I guess we could also keep it on their but not sure that's best, e.g. duplicating `id`s etc)